src/components/Body/index.tsx

```diff
@@ -1,72 +1,34 @@
-import React, { useMemo, useState } from 'react'
+import React from 'react'
 
 import { Film, TopBar } from '../../pages/MainPage/components'
-import { genreLabels } from '../../GlobalConstants'
+import { BodyProps } from '../../types'
+import { useAppDispatch, useAppSelector } from '../../hooks'
+import { setGenreOption, setSortOption } from '../../actions'
 
 import './styles.scss'
 
 const sortItems = ['RELEASE DATE', 'A-Z', 'Z-A']
 
-export interface FilmData {
-  id: number
-  imgHref: string
-  genre: string[]
-  title: string
-  releaseDate: number
-  description?: string
-  rating?: number
-  runtime?: number
-}
-
-export interface BodyProps {
-  films: FilmData[]
-  onEditVideo: () => void
-  onDeleteVideo: (ind: number) => void
-  selectedFilm: FilmData | null
-  setSelectedFilm: React.Dispatch<React.SetStateAction<any>>
-}
-
 const Body: React.FC<BodyProps> = (props) => {
+  const dispatch = useAppDispatch()
+
   const { films, onEditVideo, onDeleteVideo, selectedFilm, setSelectedFilm } = props
 
-  const [genre, setGenre] = useState(genreLabels[0])
-  const [sortOption, setSortOption] = useState(0)
-  
-  const currFilms = useMemo(() => {
-    let tempFilms = films
-    
-    if (genre !== genreLabels[0]) {
-      tempFilms = tempFilms.filter((film) => film.genre.map((g) => g.toLowerCase()).includes(genre.toLowerCase()))
-    }
-    switch (sortOption) {
-      case 0: {
-        tempFilms = tempFilms.sort((a, b) => a.releaseDate - b.releaseDate)
-        break
-      }
-      case 1: {
-        tempFilms = tempFilms.sort(({ title: a }, { title: b }) => (a < b) ? -1 : (b < a) ? 1 : 0)
-        break
-      }
-      case 2: {
-        tempFilms = tempFilms.sort(({ title: a }, { title: b }) => (a < b) ? 1 : (b < a) ? -1 : 0)
-        break
-      }
-    }
-    return tempFilms
-  }, [films, genre, sortOption])
+  const sortOption = useAppSelector(({ films }) => films.selectedSortOption)
+  const genre = useAppSelector(({ films }) => films.selectedGenre)
 
   return (
     <div className={'main-body'}>
       <TopBar
         genre={genre}
-        setGenre={setGenre}
+        setGenre={(g) => dispatch(setGenreOption(g))}
         currentSortItem={sortOption}
-        setSortItem={setSortOption}
+        setSortItem={(o) => dispatch(setSortOption(o))}
```

Comment:
that would be great if you give proper names to variables that you pass to functions instead of just letters...